radix-engine/src/blueprints/account/blueprint.rs

```diff
@@ -762,52 +710,112 @@ impl AccountBlueprint {
     }
 
     fn is_deposit_allowed<Y>(
-        deposits_mode: &AccountDepositsMode,
         resource_address: &ResourceAddress,
         api: &mut Y,
     ) -> Result<bool, RuntimeError>
     where
         Y: ClientApi<RuntimeError>,
     {
-        match deposits_mode {
-            AccountDepositsMode::AllowAll => Ok(true),
-            AccountDepositsMode::AllowList(ref allow_list)
-                if allow_list.contains(resource_address) =>
-            {
-                Ok(true)
-            }
-            AccountDepositsMode::DisallowList(ref disallow_list)
-                if !disallow_list.contains(resource_address) =>
-            {
-                Ok(true)
-            }
-            // Case: Only if the resource exists (not just that we have a vault for it). So,
-            // we need to check how much of it we have. If it's more than zero then we allow
-            // it.
-            AccountDepositsMode::AllowExisting => {
-                // Case: Deposit of XRD is always allowed in `AllowExisting` mode.
-                if *resource_address == RADIX_TOKEN {
-                    Ok(true)
+        // Read the account's current deposit operation mode
+        let default_deposit_rule = Self::get_current_default_deposit_rule(api)?;
+        let resource_deposits_configuration =
+            Self::get_resource_deposit_configuration(resource_address, api)?;
+
+        let is_deposit_allowed = match (default_deposit_rule, resource_deposits_configuration) {
```

Comment:
I would consider refactoring the implementation here, by using nested matches instead of flattened matches, because loading the default deposit rule substate is optional.

```rust
match Self::get_resource_deposit_configuration(resource_address, api)? {
    Allowed => yes,
    Disallowed = no,
    Neither => match Self::get_current_default_deposit_rule(api)? {
            Accept => yes,
            Reject => no,
            AllowExisting => additional_check()
    }
}
```

Reply:
Agreed. I think that would be a nice clean up.